04-channels/solutions/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+// Write a function that adds the contents of the slice
+// and sends the result to the result channel
+func Add(values []int, result chan<- int) {
+	total := 0
+	for _, value := range values {
+		total += value
+	}
+	result <- total
+}
+
+// Write a function that partitions the slice into
+// partitions of size partitionSize
+// Ex:
+// 	values: [1, 2, 3, 4, 5, 6, 7]
+// 	partitionSize: 3
+//  result: [[1, 2, 3], [4, 5, 6], [7]]
+func PartitionSlice(values []int, partitionSize int) [][]int {
+	partitions := [][]int{}
+	length := len(values)
+	for i := 0; i < length; i += partitionSize {
+		end := i + partitionSize
+		if end >= length {
+			end = length
+		}
+		partitions = append(partitions, values[i:end])
```

Comment:
Remind me that I'd better not forget to include slice indice selection in the slides.